api/v1alpha1/status_test.go

```diff
@@ -0,0 +1,485 @@
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonApplicationCreated,
+					Message: ConditionMessageApplicationCreated,
+				},
+			}}),
+			givenErr: fmt.Errorf(mockErrorMessage),
+			wantConditions: []metav1.Condition{
+				metav1.Condition{
+					Type:    string(ConditionApplicationReady),
+					Status:  metav1.ConditionFalse,
+					Reason:  ConditionReasonApplicationCreationFailed,
+					Message: mockErrorMessage,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			actualConditions := MakeApplicationReadyCondition(tt.givenEventingAuth, tt.givenErr)
+			// then
+			require.True(t, ConditionsEqual(tt.wantConditions, actualConditions))
+		})
+	}
+}
+
+func Test_MakeSecretReadyCondition(t *testing.T) {
+	tests := []struct {
+		name              string
+		givenEventingAuth *EventingAuth
+		givenErr          error
+		wantConditions    []metav1.Condition
+	}{
+		{
+			name: "Should application ready condition be added",
+			givenEventingAuth: createEventingAuthWith(EventingAuthStatus{Conditions: []metav1.Condition{
+				metav1.Condition{
+					Type:    string(ConditionApplicationReady),
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonApplicationCreated,
+					Message: ConditionMessageApplicationCreated,
+				},
+			}}),
+			wantConditions: []metav1.Condition{
+				metav1.Condition{
+					Type:    string(ConditionApplicationReady),
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonApplicationCreated,
+					Message: ConditionMessageApplicationCreated,
+				},
+				metav1.Condition{
+					Type:    string(ConditionSecretReady),
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonSecretCreated,
+					Message: ConditionMessageSecretCreated,
+				},
+			},
+		},
+		{
+			name: "Should update not ready secret condition to ready condition",
+			givenEventingAuth: createEventingAuthWith(EventingAuthStatus{Conditions: []metav1.Condition{
+				metav1.Condition{
+					Type:    string(ConditionApplicationReady),
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonApplicationCreated,
+					Message: ConditionMessageApplicationCreated,
+				},
+				metav1.Condition{
+					Type:    string(ConditionSecretReady),
+					Status:  metav1.ConditionFalse,
+					Reason:  ConditionReasonSecretCreationFailed,
+					Message: mockErrorMessage,
+				},
+			}}),
+			wantConditions: []metav1.Condition{
+				metav1.Condition{
+					Type:    string(ConditionApplicationReady),
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonApplicationCreated,
+					Message: ConditionMessageApplicationCreated,
+				},
+				metav1.Condition{
+					Type:    string(ConditionSecretReady),
+					Status:  metav1.ConditionTrue,
+					Reason:  ConditionReasonSecretCreated,
+					Message: ConditionMessageSecretCreated,
+				},
+			},
+		},
+		{
+			name: "Should update not ready secret condition to ready condition",
```

Comment:
The name of the test should be updated to match the test case.
```suggestion
			name: "Should update ready secret condition to ready not ready condition when error occurs",
```